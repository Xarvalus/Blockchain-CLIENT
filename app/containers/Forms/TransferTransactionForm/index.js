import * as React from 'react'
import { Field, reduxForm } from 'redux-form/immutable'
import FileField from "../CreateTransactionForm/FileField";

type Props = { handleSubmit: Function }

@reduxForm({
  form: 'createAssetTransaction',
})
export default class TransferTransactionForm extends React.PureComponent<Props> {
  render(): React.Node {
    const { handleSubmit } = this.props

    return (
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <label htmlFor="transactionId">Transaction ID</label>
          <Field type="text" className="form-control" name="transactionId" component="input" />
        </div>

        <div className="form-group">
          <label htmlFor="note">Note</label>
          <Field type="text" className="form-control" name="note" component="input" />

          <label htmlFor="price">Price</label>
          <Field type="text" className="form-control" name="price" component="input" />

          <label htmlFor="auctionHouse">Auction House</label>
          <Field type="text" className="form-control" name="auctionHouse" component="input" />
        </div>

        <div className="form-group clearfix">
          <button type="submit" className="btn btn-primary mb-2 float-right">Submit</button>
        </div>
      </form>
    )
  }
}
