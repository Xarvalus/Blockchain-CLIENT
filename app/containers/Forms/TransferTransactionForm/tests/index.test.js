import React from 'react'
// import { shallow } from 'enzyme'
import ShallowRenderer from 'react-test-renderer/shallow'

import TransferTransactionForm from '../index'

describe('<TransferTransactionForm />', () => {
  // let wrapper
  // beforeEach(() => {
  //   wrapper = shallow(<TransferTransactionForm />)
  // })

  it('renders correctly', () => {
    const renderer = new ShallowRenderer()
    const snapshot = renderer
      .render(<TransferTransactionForm />)

    expect(snapshot).toMatchSnapshot()
  })
})
