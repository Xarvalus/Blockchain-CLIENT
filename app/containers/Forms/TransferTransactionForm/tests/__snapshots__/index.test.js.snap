// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransferTransactionForm /> renders correctly 1`] = `
<Connect(Form(TransferTransactionForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="createAssetTransaction"
  getFormState={[Function]}
  initialValues={undefined}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
